airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryGcsDestinationTest.java

```diff
@@ -299,7 +299,6 @@ void testWriteFailure() throws Exception {
 
     assertThrows(RuntimeException.class, () -> consumer.accept(spiedMessage));
     consumer.accept(MESSAGE_USERS2);
-    assertThrows(RuntimeException.class, () -> consumer.close()); // check if fails when data was not loaded to GCS bucket by some reason
```

Comment:
Why is this assertion removed? Just want to make sure that it is not removed by accident.

Reply:
There were two different implementations.
GCS impl throws Runtime exception if it closed with failed flag.
Common impl don't throw.
The new impl stops retrieving messages and start close operation with corresponding Fail flag. In this case we don't apply already loaded records and clean everything in a controlled way.